api/app/src/providers/withings.ts

```diff
@@ -79,6 +76,60 @@ export class Withings extends Provider implements OAuth2 {
     return JSON.stringify(response.data.body);
   }
 
+  async getAccessToken(connectedUser: ConnectedUser): Promise<string> {
+    const providerData = getProviderDataFromConnectUserOrFail(connectedUser, PROVIDER_WITHINGS);
+
+    const token = providerData.token;
+
+    const refreshedToken = await this.checkRefreshToken(token, connectedUser);
+
+    return refreshedToken.access_token;
+  }
+
+  async checkRefreshToken(token: string, connectedUser: ConnectedUser): Promise<Token> {
+    const access_token = JSON.parse(token)
+    const now = dayjs(Date.now()).unix()
+    const isExpired = access_token.expires_at < now;
```

Comment:
Should we use the same logic as the simple oauth2 lib? They use an expiration window so the expiry is caught a bit earlier